### PR TITLE
Update minimum Pandas version to 1.5.0

### LINKED ISSUE #3759: Update for Pandas 1.5
We should upgrade to Pandas 1.5 support, possibly as a minimum version as the STLDecomposer references Int64Index.  Let's update the pandas version and get the tests passing.

### PR DESCRIPTION
**Pull request by christopherbunn**
Resolves #3759

### CODE REVIEW

File: .github/meta.yaml [hunk 1]
```diff
@@ -73,7 +73,7 @@ outputs:
         - lime >=0.2.0.1
         - python >=3.8.*
         - imbalanced-learn >=0.9.1
-        - sktime <0.13.1,>=0.12.0
+        - sktime >=0.13.3
```
Comment (christopherbunn) [inline .github/meta.yaml:76]: The conda is currently failing as it looks like the [latest version of sktime](https://anaconda.org/conda-forge/sktime) that is available on conda is 0.12.1 when we need this version for pandas 1.5 support. If pandas 1.5 support isn't super high priority we might need to skip conda support for now.

Comment (jeremyliweishih) [reply .github/meta.yaml:76]: https://github.com/sktime/sktime/issues/2921 - related. Can you file an issue against sktime to release 0.13.3?

**Review by jeremyliweishih — APPROVED** [review]
This LGTM pending conda release of sktime.

**Review by eccabay — APPROVED** [review]
🚢

**Pull request merged**